base/abstractarray.jl

```diff
@@ -170,31 +170,42 @@ _length(A) = (@_inline_meta; length(A))
 
 """
     lastindex(collection) -> Integer
+    lastindex(collection, d) -> Integer
 
-Return the last index of the collection.
+Return the last index of `collection`, optionally constrained to be within dimension `d`.
```

Comment:
Thanks! "contrain" sounds weird here, doesn't it? After reading it written that way, I didn't understand what the function did anymore. ;-)

Maybe something like "If `d` is provided, return the last cartesian coordinate for that dimension."? (Or does "cartesian coordinate" only refer to the tuple of indices?)

Reply:
Fair.  How about:

> Return the last index of `collection`. If `d` is given, return the last index of `collection` along dimension `d`.